pkg/config/config.go

```diff
@@ -450,7 +450,11 @@ func (dc *DeploymentConfig) SetCLIVariables(cliVariables []string) error {
 			return fmt.Errorf("invalid format: '%s' should follow the 'name=value' format", cliVar)
 		}
 
-		key, value := arr[0], arr[1]
+		// Convert the variable's string litteral to its equivalent default type.
+		var out interface{}
+		yaml.Unmarshal([]byte(arr[1]), &out)
```

Comment:
Maybe a silly question, what happens if this fails? should you not check for an error?